lib/ukstore/include/uk/store.h

```diff
@@ -29,3 +29,404 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
+#ifndef __UK_STORE_H__
+#define __UK_STORE_H__
+
+#include <string.h>
+#include <uk/config.h>
+#include <uk/assert.h>
+#include <uk/list.h>
+#include <uk/arch/atomic.h>
+#include <uk/essentials.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !CONFIG_LIBUKSTORE
+
+#define uk_store_libid(libname) (-1)
+
+#define uk_store_dynamic_entry_create(folder, name, type, get, set)	\
+	do { } while (0)
+
+#define uk_store_dynamic_folder_create(name)	\
+	do { } while (0)
+
+#define uk_store_add_folder(folder)	\
+	do { } while (0)
+
+#define uk_store_free_folder(folder)	\
+	do { } while (0)
+
+#define uk_store_get_folder(library_id, folder_name)	\
+	do { } while (0)
+
+#define uk_store_get_entry(libname, foldername, entryname)	\
+	do { } while (0)
+
+#define uk_store_free_entry(entry)	\
+	do { } while (0)
+
+#define uk_store_update_getter(entry, e_type, func)	\
+	do { } while (0)
+
+#define uk_store_update_setter(entry, e_type, func)	\
+	do { } while (0)
+
+#define uk_store_get_value(entry, outtype, out)	\
+	do { } while (0)
+
+#define uk_store_set_value(entry, intype, in)	\
+	do { } while (0)
+
+#else /* !CONFIG_LIBUKSTORE */
+
+/* Library-specific information generated at compile-time */
+#include <uk/_store_libs.h>
+
+/* Do not call directly */
+#define _uk_store_libid(libname)	\
+	__UK_STORE_ ## libname
+
+/**
+ * Returns the id of a given library.
+ *
+ * @param libname the library to search for (not a string, e.g. LIBVFSCORE)
+ * @return the id
+ */
+#define uk_store_libid(libname)	\
+	_uk_store_libid(libname)
+
+#endif /* CONFIG_LIBUKSTORE */
+
+/**
+ * Returns the id of the place it was called from.
+ *
+ * @return this libraries' id
+ */
+#define uk_store_libid_self()	\
+	uk_store_libid(__LIBNAME__)
+
+/* All basic types that exist - use these when interacting with the API */
+enum uk_store_entry_type {
+	_LIB_UK_STORE___undef = 0,
+	_LIB_UK_STORE___s8    = 1,
+	_LIB_UK_STORE___u8    = 2,
+	_LIB_UK_STORE___s16   = 3,
+	_LIB_UK_STORE___u16   = 4,
+	_LIB_UK_STORE___s32   = 5,
+	_LIB_UK_STORE___u32   = 6,
+	_LIB_UK_STORE___s64   = 7,
+	_LIB_UK_STORE___u64   = 8,
+	_LIB_UK_STORE___uptr  = 9,
+	_LIB_UK_STORE___charp = 10
+};
+
+/* Transforms a simple data type to a ukstore specifice one */
+#define uk_store_get_entry_type(simple_type)	\
+	(_LIB_UK_STORE___ ## simple_type)
```

Comment:
If you remove the update getter&setter you can move this definiton into the C file

Reply:
This is also used by `__UK_STORE_STATIC_ENTRY` in the header. If I move it that will stop working